### DISCUSSION TITLE
How to skip first first run of queryFn?

### DISCUSSION THREAD

**Discussion by bendvc** [TanStack/query#4212] [Q&A]
I am doing SSR in a project that I'm working on and would like to skip the first run of any query that is being hydrated. Is there a standard way of doing this? It would be nice if the `enabled` prop could accept a function that I could pass a `falseThenTrue` type function too. Is there an obvious solution that I'm missing?

TIA

UPDATE:

For clarification this is what is happening.

1. **On the server**: We use react-ssr-prepass to help await on queries so we can dehydrate the client and serialize the content. 
2. **On the client**: The app is hydrated and the react query client is also hydrated.
3. **On the client**: When the app is mounted, the same queries that ran on the server are then run again. I would like to avoid running these queries on the initial mount of the application.

**Comment by TkDodo** [comment]
I don't understand the question, I don't think you're providing enough context. I'm just guessing:

- you're rending a query on the server with getServerSideProps
- you're using `Hydrate` from react-query on the client and `dehydrate` on the server
- the query still fetches on the client

is that right?

**Reply by bendvc** [reply]
For the post part yes. I am not using nextjs. But either way, we render our app on the server where we use `react-ssr-prepass` and await all the queries. On the client we hydrate our application and our react query client. The thing I am trying to avoid is the same queries that I ran on the server are again being run on the client when the app is mounted. I would ideally like to skip that first fetch but on subsequent fetches do fetches as you normally would. 

I have been playing with the `refetchOnMount` option. It might hold the key to my answer.

**Reply by TkDodo** [reply]
have you read: https://tanstack.com/query/v4/docs/guides/ssr#staleness-is-measured-from-when-the-query-was-fetched-on-the-server
